### PR TITLE
functionality to store dependency snapshot in database

### PR DESCRIPTION
**Pull request by ravsa**
- store dependency snapshot in database or update existing snapshot
    `/api/v1/stack-analyses?persist=<true|false>&requestId=<request_id>`
- minor bug fixes

### CODE REVIEW

**Comment by humaton** [comment]
[test]

File: bayesian/api_v1.py [hunk 1]
```diff
@@ -919,35 +924,53 @@ def post():
...
             _session.post('{}/api/v1/stack_aggregator'.format(api_url), json=deps)
             _session.post('{}/api/v1/recommender'.format(api_url), json=deps)
         except Exception as exc:
             raise HTTPError(500, ("Could not process {t}.".format(t=request_id))) from exc
-
-        return {"status": "success", "submitted_at": str(dt), "id": str(request_id)}
+        if persist:
+            if sid:
+                try:
+                    result = rdb.session.query(StackAnalysisRequest)\
```
Comment (miteshvp) [inline bayesian/api_v1.py]: You can also look at http://docs.sqlalchemy.org/en/latest/dialects/postgresql.html?highlight=conflict#insert-on-conflict-upsert

**Comment by miteshvp** [comment]
@ravsa - Let's discuss this workflow again. Although it looks good, but please change this PR in WIP till then

File: bayesian/api_v1.py (hunk 1 above)
Comment (ravsa) [reply bayesian/api_v1.py]: Thanks @miteshvp, I will do the changes.

**Comment by ravsa** [comment]
sure, current workflow 
1) If no parameters passed API will work as it was working before.
2) If `persist=true`  and `sid` is passed then it will update the dep-snapshot.
3) if `persist=false` is passed as a parameter then no insertion and update in the database.

**Comment by miteshvp** [comment]
> If persist=true and sid is passed then it will update the dep-snapshot.

We just need to upsert in API server without any parameters actually needed (based on sid present or not).
`persist=false` will help backbone to take a decision whether to persist the record or not

**Pull request merged**